core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -153,14 +153,32 @@ public long snapshotId() {
     }
   }
 
-  class SetCurrentSnapshot implements MetadataUpdate {
-    private final Long snapshotId;
+  class RemoveSnapshotRef implements MetadataUpdate {
```

Comment:
`RemoveSnapshotRef` is used for dropping a branch instead of `SetCurrentSnapshot(null)`. Now the snapshot can be required instead of nullable.